File: src/c/ch01/1.2/l2/demo_multiplication.c
#include <stdio.h>

int naive_mul(int x, int y) {
    return x * y; // just builtin
}

int main(void) {
    int a = 1234, b = 5678;
    printf("Naive multiplication: %d\n", naive_mul(a, b));
    printf("Builtin multiplication: %d\n", a * b);
    return 0;
}

File: src/go/ch01/1.2/l2/demo_multiplication.go
package main

import "fmt"

// naiveMul just uses Go's built-in operator as a baseline.
func naiveMul(x, y int) int { return x * y }

func main() {
	a, b := 1234, 5678
	fmt.Println("Naive multiplication:", naiveMul(a, b))
	fmt.Println("Builtin multiplication:", a*b)

	// Run other demos
	runKaratsubaDemo()
	runModexpDemo()
	runExercises()
}
